net/tshttpproxy/tshttpproxy_synology.go

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build linux
+// +build linux
+
+package tshttpproxy
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"sync"
+	"time"
+
+	"tailscale.com/util/lineread"
+	"tailscale.com/version/distro"
+)
+
+func init() {
+	if distro.Get() == distro.Synology {
```

Comment:
rather than do this disto.Get() work at init-time, I'd prefer you just do `sysProxyEnv = linuxSysProxyEnv` and then do this `distro.Get()` lazily if/when the proxy call is ever made